### PR TITLE
Even more dynamic fixes.

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
Sat on this for a day but nothing else has cropped up. Seems we might be nearing the end of the obviously broken stuff.

## Why It's Good For The Game
bug fix

## Changelog
:cl: Skoglol
fix: Dynamic: Scaling now even more accurate.
admin: Dynamic: Fixed false positive configuration warning.
fix: Dynamic: Roundstart revs should work again.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/dynamic/dynamic_rulesets.dm [hunk 1]
```diff
@@ -111,7 +111,7 @@
 		return (threat_level >= high_population_requirement)
 	else
 		pop_per_requirement = pop_per_requirement > 0 ? pop_per_requirement : mode.pop_per_requirement
-		if(antag_cap && requirements.len != antag_cap.len)
+		if(antag_cap.len && requirements.len != antag_cap.len)
```
Comment (TheChosenEvilOne) [inline code/game/gamemodes/dynamic/dynamic_rulesets.dm]: won't this runtime because antag_cap is null?

Comment (kriskog) [reply code/game/gamemodes/dynamic/dynamic_rulesets.dm]: Its not null, thats the issue with the old check. It is initialized as a list
var/list/antag_cap = list()

I checked with MoMMI this time
image

**Comment by kriskog** [comment]
travis please

File: code/game/gamemodes/dynamic/dynamic_rulesets.dm (hunk 1 above)
Comment (TheChosenEvilOne) [reply code/game/gamemodes/dynamic/dynamic_rulesets.dm]: You removed the default initialization above, so it will be null. no?

Comment (kriskog) [reply code/game/gamemodes/dynamic/dynamic_rulesets.dm]: I must be delirious, I dont remember doing that :eyes:

Comment (kriskog) [reply code/game/gamemodes/dynamic/dynamic_rulesets.dm]: Thanks, fixed

**Comment by TheChosenEvilOne** [comment]
Latejoin Revs instantly loose because they start outside the station z-level.

**Comment by kriskog** [comment]
> 
> 
> Also should change how postsetup works to prevent runtimes, just put everything on a timer and make it call a proc that adds them to the current_rules list and refunds the ruleset if execute fails.

Is there a particular runtime we are trying to prevent?

**Comment by TheChosenEvilOne** [comment]
Runtimes caused by calling rule_process before execute, revolution does this

**Pull request merged**